background.js

```diff
@@ -22,9 +28,34 @@ function(details) {
 			details.requestHeaders.push({name: name, value: value});
 		});	
 
+		if (profile.aws) {
+			const awsHeaders = awsSigner.beforeSendHeaders(details, profile.aws);
+			if (awsHeaders) {
+				awsHeaders.forEach(header => details.requestHeaders.push(header));
+			}
+		}
+
 		return {requestHeaders: details.requestHeaders};
 	}
 },
 {urls: ["<all_urls>"]},
 ["blocking", "requestHeaders"]);
 
+const isUrlValid = (url) => {
```

Comment:
Can we move these const declarations to the top of the file please? I feel it would make it more readable